### ISSUE TITLE
Slim 3 Upgrade Guide Missing Information

### ISSUE THREAD

**Issue by kwhat** [slimphp/Slim-Website#24]
Hi,

I noticed that the upgrade guide for slim3 does not include information about `Slim\Slim` changing to `Slim\App` or what happened to `Slim::registerAutoloader()`.  

I also can't figure out what happened to the `$app->container->singleton(...)` stuff.  I assume that got changed to `$app->getContainer()->offsetSet(...)`

Looks like `$app->configureMode(...)` has also changed.

Migrating strategy for PrettyExceptions.

`Route::setDefaultConditions(...)` has changed or was removed?

Thanks.

**Comment by silentworks** [comment]
Will add these to the Upgrade guide.

`Slim::registerAutoloader()` have been removed.
`$app->container->singleton(...)` is now `$app['...'] = function () {};` - read Pimple docs for more info
`$app->configureMode(...)` have been removed.

PrettyExceptions cause lots of issues for many people, so this have been removed.

`Route::setDefaultConditions(...)` is no longer part of Slim, we have decided to use FastRoute as our default router and don't require this feature anymore.
